lib/bolt/target.rb

```diff
@@ -80,6 +80,11 @@ def resources
       inventory_target.resources
     end
 
+    def resource(type, title)
```

Comment:
This logic shouldn't live on `Bolt::Target` as it's not part of the API of the `Target` datatype. It can either be implemented directly in the Puppet function (if all we're doing is calling `resources["<ref>"]`) or it should follow the pattern of similar functions, which is to call a method on the inventory that delegates to `Bolt::Inventory::Target`.

Alternately, we could just implement this as a function on the `Target` datatype.